tasks/main.yml

```diff
@@ -16,21 +16,8 @@
   with_items: '{{ boot_config_lines }}'
   register: _boot_config_lines
 
-- name: restart machine  # noqa 503
-  shell: sleep 2 && shutdown -r now "/boot/config.txt config changed"
-  async: 1
-  poll: 0
-  ignore_errors: true
-  when: _boot_config.changed or _boot_config_lines.changed
-
-
-- name: waiting for machine to come back  # noqa 503
-  delegate_to: localhost
-  connection: local
-  wait_for:
-    port: 22
-    host: "{{ ansible_ssh_host|default(ansible_host)|default(inventory_hostname) }}"
-    delay: 20
-    timeout: 120
-  become: false
+- name: "restart machine"  # noqa 503
+  reboot:
+    msg: "Reboot by Ansible, because /boot/config.txt config changed."
+    reboot_timeout: 300   # (= 5 minutes)
```

Comment:
is it intentional to remove the when condition?
aka
```
  when: _boot_config.changed or _boot_config_lines.changed
```

Reply:
Yes, otherwise there would be two of them for a single task. ;-)
I replaced two tasks for a single task. 

Perhaps you overlooked the last line of the file? (Like I did the `.travis.yml`).
The `when:` on former line 24 is indeed removed intentionally.
The `when:` on former line 36, now line 23, remains.

Have a look at the [file itself](https://github.com/infothrill/ansible-role-rpi-boot-config/blob/4ef36f7c1d024478a650c58fa33fdea986f2e933/tasks/main.yml). It will probably look a lot clearer than the diff.